### PR TITLE
Clean Mac Tests Failures

### PR DESCRIPTION
**Pull request by amystamile-usgs**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation change (update to the documentation; no code change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Infrastructure change (changes to things like CI or the build system that do not impact users)

## Checklist:

- [ ] I have read and agree to abide by the [Code of Conduct](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/Code-Of-Conduct.md)
- [ ] I have read the [**CONTRIBUTING**](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation and I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] I have added myself to the [.zenodo.json](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/.zenodo.json) document.
- [ ] I have added any user impacting changes to the [CHANGELOG.md](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CHANGELOG.md) document.

## Licensing
This project is mostly composed of free and unencumbered software released into the public domain, and we are unlikely to accept contributions that are not also released into the public domain. Somewhere near the top of each file should have these words:

> This work is free and unencumbered software released into the public domain. In jurisdictions that recognize copyright laws, the author or authors of this software dedicate any and all copyright interest in the software to the public domain.

- [ ] I dedicate any and all copyright interest in this software to the public domain. I make this dedication for the benefit of the public at large and to the detriment of my heirs and successors. I intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.

### CODE REVIEW

File: isis/tests/FunctionalTestsHimos.cpp [hunk 1]
```diff
@@ -62,7 +62,7 @@ TEST_F(MroHiriseCube, FunctionalTestHimosDefault) {
     StopTime                  = 2006-09-29T15:16:35.036
     SpacecraftClockStopCount  = 844010213:55196
     IncidenceAngle            = 59.687930340662 <DEG>
-    EmissionAngle             = 0.091672512443932 <DEG>
+    EmissionAngle             = 0.091672512439956 <DEG>
```
Comment (acpaquette) [inline isis/tests/FunctionalTestsHimos.cpp]: This is now failing for linux as seen in the nightly builds:
https://jenkins.prod-asc.chs.usgs.gov:8443/job/ISIS/view/change-requests/job/PR-5082/

File: isis/tests/TgoCassisModuleTests.cpp [hunk 2]
```diff
@@ -2765,7 +2754,7 @@ TEST_F(TgoCassisModuleKernels, TgoCassisTestProjSingleStitchedFrame) {
   Histogram *hist = exportCube.histogram();
 
   EXPECT_NEAR(hist->Average(), 0.26625623495550205, 0.0001);
-  EXPECT_NEAR(hist->Sum(), 444615.96850222297, 0.0001);
-  EXPECT_EQ(hist->ValidPixels(), 1669880);
+  EXPECT_NEAR(hist->Sum(), 444615.69537125883, 0.0001);
+  EXPECT_EQ(hist->ValidPixels(), 1669879);
```
Comment (acpaquette) [inline isis/tests/TgoCassisModuleTests.cpp]: Same as above

**Review by acpaquette — CHANGES_REQUESTED** [review]
Some of the changed tests are now failing on linux, for the pvl comparison the best way I could think to get around it is to add a tolerance when comparing integers. I am unsure on how difficult that would be but it is worth checking.

For the TGO Cassis Module tests I don't know what would be a good solution. It might be worth digging in a bit to see why we are getting one less valid pixel

File: isis/tests/FunctionalTestsHimos.cpp (hunk 1 above)
Comment (amystamile-usgs) [reply isis/tests/FunctionalTestsHimos.cpp:65]: I am wondering if we should avoid checking against the label this way. It does not allow us to compare within a certain decimal range.

Comment (amystamile-usgs) [reply isis/tests/FunctionalTestsHimos.cpp]: I will work on getting these switched to individual asserts.

**Comment by amystamile-usgs** [comment]
For the TgoCassisTestProjSingleStitchedFrame test: 

The Mac output cube is picking up an extra Null Pixel causing the ValidPixels to be off by one and therefore causing a different sum. 

This is the result of running cubediff between the mac and linux outputs of this test:

```
Group = Results
  Compare                   = Different
  Sample                    = 1260
  Line                      = 18
  Band                      = 1
  AverageDifference         = 5.16362740619162e-07
  StandardDeviation         = 2.85657834831671e-05
  Variance                  = 8.16003986007184e-10
  MinimumDifference         = 3.72529029846191e-09
  MaximumDifference         = 0.0046229721046984
  MaxDifferenceSample       = 1348
  MaxDifferenceLine         = 1876
  MaxDifferenceBand         = 1
  ValidPixelDifferences     = 124524
  SpecialPixelDifferences   = 1
  SigFigAccuracy            = 0
  SigFigMaxDifferenceSample = 1348
  SigFigMaxDifferenceLine   = 1876
  SigFigMaxDifferenceBand   = 1
End_Group
```

**Comment by amystamile-usgs** [comment]
@acpaquette Are you seeing failures with TgoCassisTestProjSingleStitchedFrame on your older Mac OS?

**Review by acpaquette — APPROVED** [review]

**Pull request merged**